Lib/fontTools/feaLib/builder.py

```diff
@@ -1046,23 +1073,31 @@ def add_mark_mark_pos(self, location, baseMarks, marks):
     def add_class_pair_pos(self, location, glyphclass1, value1,
                            glyphclass2, value2):
         lookup = self.get_lookup_(location, PairPosBuilder)
-        lookup.addClassPair(location, glyphclass1, value1, glyphclass2, value2)
+        v1 = makeOpenTypeValueRecord(value1, pairPosContext=True)
+        v2 = makeOpenTypeValueRecord(value2, pairPosContext=True)
+        lookup.addClassPair(location, glyphclass1, v1, glyphclass2, v2)
 
     def add_subtable_break(self, location):
-        self.cur_lookup_.add_subtable_break(location)
+        self.cur_lookup_.add_subtable_break(self.format_location_(location))
 
     def add_specific_pair_pos(self, location, glyph1, value1, glyph2, value2):
         lookup = self.get_lookup_(location, PairPosBuilder)
-        lookup.addGlyphPair(location, glyph1, value1, glyph2, value2)
+        v1 = makeOpenTypeValueRecord(value1, pairPosContext=True)
+        v2 = makeOpenTypeValueRecord(value2, pairPosContext=True)
+        lookup.addGlyphPair(location, glyph1, v1, glyph2, v2)
 
     def add_single_pos(self, location, prefix, suffix, pos, forceChain):
         if prefix or suffix or forceChain:
             self.add_single_pos_chained_(location, prefix, suffix, pos)
         else:
             lookup = self.get_lookup_(location, SinglePosBuilder)
             for glyphs, value in pos:
+                otValueRecord = makeOpenTypeValueRecord(value, pairPosContext=False)
                 for glyph in glyphs:
-                    lookup.add_pos(location, glyph, value)
+                    try:
+                        lookup.add_pos(location, glyph, otValueRecord)
+                    except OpenTypeLibError as e:
+                        raise FeatureLibError("Failed to build", location) from e
```

Comment:
again here, "Failed to build" sounds quite generic, may be better to add more context.
Or is it that you are relying on that `if self.__context__: ...` clause in `FeatureLibError.__str__` method, to basically ignore this generic message and use the one from the underlying exception?
I don't like this approach, as I pointed out earlier.